builtin/mainmenu/modmgr.lua

```diff
@@ -438,25 +438,56 @@ function modmgr.preparemodlist(data)
 
 	local worldfile = Settings(filename)
 
+	modmgr.mods_by_name = {}
+	local mods_active_by_name = {}
 	for key,value in pairs(worldfile:to_table()) do
 		if key:sub(1, 9) == "load_mod_" then
 			key = key:sub(10)
 			local element = nil
 			for i=1,#retval,1 do
```

Comment:
Possibly less messy by iterating through `retval`, checking against the value of `worldfile:get("load_mod_" .. this_mod.name)` (AFAIK, returns empty string if it wasn't found).

Reply:
The code wasn't prepared to handle modpacks, and when the modpack addition was hacked in, it became less manageable. I worked with what I had. I agree that the whole code around mod handling needs a rewrite, as I noted [here](https://github.com/minetest/minetest/issues/4183#issuecomment-355832652), but that's out of scope for this PR.

If we want to backport this fix, the amount of new, untested code should be kept to a minimum.

Reply:
Each new feature makes it harder and more annoying to rewrite it, that's basically my concern. I see that it doesn't belong directly into this PR but I'm afraid that the rewrite won't happen any soon due "don't touch a running system".